nodes/delete_node.js

```diff
@@ -1,30 +1,87 @@
 module.exports = function (RED) {
+  'use strict';
+
   const { WazoApiClient } = require('@wazo/sdk');
+  
+  function setProperty(node, msg, name, type, value) {
+      if (type === 'msg') {
+          msg[name] = value;
+      } else if (type === 'flow') {
+          node.context().flow.set(name, value);
+      } else if (type === 'global') {
+          node.context().global.set(name, value);
+      }
+  }
+
+  function sendErrorMessage(node, text) {
+      node.status({
+          fill: 'red',
+          shape: 'ring',
+          text: (typeof text === "object") ? JSON.stringify(text) : text
+      });
+  }
+  
+  function setStatus(node, text, color, shape) {
+    if (!text && !color && !shape) {
+      node.status({});
+    } else {
+      node.status({
+        fill: color || 'blue',
+        shape: shape || 'ring',
+        text: text
+      });
+    }
+  }
+  
+  function checkType(node, parameter, type) {
+    if (typeof parameter !== type) {
+      const text = RED._('delete_node.errors.falsetype') + type;
+      sendErrorMessage(node, text);
+      throw(text);
+    }
+  }
+
 
   function delete_node(n) {
-    RED.nodes.getNode(this, n);
+    RED.nodes.createNode(this, n);
     conn = RED.nodes.getNode(n.server);
     this.client = conn.client.application;
+    
+    let node = this;
+    setStatus(node);
+
+    node.callnode = n.callnode;
+    node.callnodeType = n.callnodeType;
 
-    var node = this;
+    node.on('input', async (msg, send, done) => {
+      setStatus(node, "running");
 
-    node.on('input', async msg => {
-      node_uuid = msg.payload.node_uuid;
-      application_uuid = msg.payload.application_uuid;
+      let callnode = RED.util.evaluateNodeProperty(node.callnode, node.callnodeType, node, msg);
+      const node_uuid = callnode.node_uuid || callnode.uuid;
+      const application_uuid = callnode.application_uuid;
+
+      try { 
+        checkType(node, node_uuid, "string");
+        checkType(node, application_uuid, "string");
+      } catch (err) {
+        done(err);
+        return;
+      }
 
       if (node_uuid && application_uuid) {
         node.log("Delete node");
         try {
-          const deleteNode = await node.client.removeNode(application_uuid, node_uuid);
-          node.log(`Remove node ${node_uuid}`);
-          msg.payload.application_uuid = application_uuid;
-          msg.payload.node_uuid = node_uuid;
-          msg.payload.data = deleteNode;
-          node.send(msg);
+          const result = await node.client.removeNode(application_uuid, node_uuid);         
+          node.log(result);
+          setStatus(node, `Last deleted: ${node_uuid}`, "green", "dot");
+          setProperty(node, msg, node.callnode, node.callnodeType, result);
```

Comment:
This is a complete new method here to send data to other node, what the impact with the other node?

Reply:
setProperty will overwrite the input defined in the parameter (which I missed to upload, see your comment above).
Example:
if msg.payload was the input, it will set msg.payload = result.

Reply:
ok i  think i understand, but i'm not sure we need to remove the application_uuid and node_uuid on the payload because we will broke the flow with the other nodes. I did that because unfortunately all events don't have all the correct information for make a correct flow. We need to fix it in wazo-platform before removing this informations. Make sense or i have something i don't have understand?

Reply:
Agreed. I'll make the new input field optional in a way, that if there is no input, it will consider application_uuid and node_uuid